### PR TITLE
feat: implement v6 option in tpc clusterizer

### PR DESCRIPTION
**Pull request by osbornjd**
This should implement @Ishangoel11 v6 cluster object and get all of the types consistent between the base class and derived class IO object.

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

## Motivation and Context
This PR introduces an optional v6 clustering mode in the TPC clusterizer to provide richer per-cluster characterization (edge/mixing/overlap) and to tighten type consistency between the `TrkrCluster` base interface and the v5/v6 derived cluster implementations. It also updates tracking diagnostics and cluster NTuples to serialize the new observables. *Note:* AI can make mistakes—please use best judgment and verify key API/IO impacts locally, with extra attention to the IO object refactoring/standardization mentioned by the author (independent verification that IO changes are correct and not unintentionally corrupted).

## Key Changes
- **TPC clustering (v6 support + richer bookkeeping)**
  - Added a `ClusterCounters`-based workflow to track cluster overlap, overall edge counts, detailed per-side edge categories, and phi/time “mix” touching indicators.
  - Implemented neighbor scanning (`check_cluster_touching`) across adjacent phi/time cells to populate mix indicators.
  - Updated cluster building (`get_cluster`, `calc_cluster_parameter`) to propagate `ClusterCounters` into the produced cluster object.
  - Added a `DetailedClusterAnalysis()`/`m_debug` debug path to select **`TrkrClusterv6`** (otherwise **`TrkrClusterv5`**).
  - Extended `ProcessSectorData` to preserve original ADC data for neighbor checks and to reset/pipeline `ClusterCounters` appropriately.

- **Cluster object API/type contract alignment**
  - **`TrkrCluster.h`**: const-correct setter parameters plus multiple public type contract changes (notably `getPadMax/getTBinMax` float→int, mix getters int→char, phi/tbin bin lo/hi float→`unsigned short`) and associated sentinel changes.
  - **`TrkrClusterv6.h`**: standardized signatures with `override` and aligned member types/defaults to the updated contracts.
  - Minor override/signature consistency updates in **v4/v5** cluster classes (const/override correctness, no intended logic changes).

- **Tracking diagnostics + ROOT output extensions**
  - **`TrackResiduals`**: read `EventHeader`, store `m_evt_id`, and write a new `evt_id` branch into multiple ROOT trees.
  - Added/expanded cluster state resets and extended cluster/edge/mix observable filling into existing output trees.

- **Cluster NTuple schema updates**
  - **`TrkrNtuplizer`**: extended `_ntp_cluster`/`n_cluster` schema and filling to include new per-cluster ADC/bin/max/phase and edge/mixing-related fields.
  - Changed `nclusize` to use `cluster->getRSize()` instead of `cluster->getSize()`.

## Potential Risk Areas
- **IO / analysis compatibility**
  - ROOT tree branch additions/renames (`TrackResiduals`) and TNtuple column/schema layout changes (`TrkrNtuplizer`) may require downstream analysis updates.
- **API/type contract changes**
  - Public method return types and sentinels in `TrkrCluster` (and v6 alignment) can break downstream code or subtly change behavior if callers assumed `float`/NaN or other sentinel semantics.
- **Reconstruction behavior differences**
  - When v6 mode is selected via the debug/detailed-analysis path, new neighbor/mix/edge logic may change cluster-derived observables versus v5.
- **Performance / resource use**
  - Additional neighbor scans and more per-cluster serialized fields can increase compute and IO load in v6/detailed mode.
- **Thread-safety / state management**
  - Ensure per-cluster/per-sector `ClusterCounters` resets and any new debug flag plumbing do not leak state across parallel execution contexts.

## Possible Future Improvements
- Add regression tests that compare representative v5 vs v6 outputs (especially boundary and bin/phase edge cases for mix/overlap/edge categories).
- Provide a short migration note for downstream users about updated `TrkrCluster` contracts and updated ROOT/TNtuple schemas.
- Profile v6 performance impact on large productions and consider configurability for neighbor/mix criteria (without code changes).

### CODE REVIEW

**Comment by osbornjd** [comment]
@pinkenburg can you give an extra set of eyes on the IO objects and make sure I haven't inadvertently mangled something, I tried cleaning them up and making them consistent

File: offline/packages/TrackingDiagnostics/TrackResiduals.cc [hunk 1]
```diff
@@ -301,6 +328,21 @@ int TrackResiduals::process_event(PHCompositeNode* topNode)
...
+    m_evt_id = eventheader->get_EvtSequence();
+  }
+  else
+  {
+    m_evt_id = -1;
+  }
+
+  auto *rcs = recoConsts::instance();
+  m_runnumber = rcs->get_IntFlag("RUNNUMBER");
+  m_segment = rcs->get_IntFlag("RUNSEGMENT");
```
Comment (pinkenburg) [inline offline/packages/TrackingDiagnostics/TrackResiduals.cc:344]: The segment is really meaningless. It doesn't go back to the rawdata (so it won't help locating the event) but is just a feature of the filename depending on how many events we store in a file which is subject to change.

**Review by pinkenburg — APPROVED** [review]
